### PR TITLE
remove libco-canonical

### PR DESCRIPTION
**Pull request by wucke13**
###### Motivation for this change

This set of patches removes libco-canonical from the nixpkgs. It was used for dqlite, lxd and some other canonical stuff. However, it is not used anymore. Canonical also deleted it's repo - hence I would like to remove it from the nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wucke13** [comment]
@GrahamcOfBorg build libraft dqlite lxd

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15588,8 +15588,6 @@ in
 
   libclxclient = callPackage ../development/libraries/libclxclient  { };
 
-  libco-canonical = callPackage ../development/libraries/libco-canonical { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15591]: Missing alias entry.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123139` at 3cfba505 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>dqlite</li>
    <li>lxd</li>
    <li>raft-canonical</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md">no-flags-spaces</a>

`buildFlags` cannot contain spaces, please use `buildFlagsArray` in Bash.

Near pkgs/tools/admin/lxd/default.nix:43:3:
```
   |
43 |   buildFlags = [ "-tags libsqlite3" ];
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/top-level/all-packages.nix:15591]: What is an "alias entry" ?

Comment (wucke13) [reply pkgs/top-level/all-packages.nix:15591]: https://github.com/NixOS/nixpkgs/commit/a6bf99d97bbbae9fb2e52faa751d792f8f888bf7 does this do what you mentioned?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15588,7 +15588,7 @@ in
...
+  libco-canonical = throw "libco-canonical removed since 2021-05-16: Canonical deleted the repo, libco-canonical is not used anymore.";
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15591]:
```suggestion
  libco-canonical = throw "libco-canonical: Canonical deleted the repo, libco-canonical is not used anymore."; # added 2021-05-16
```

and move that line to aliases.nix.

Comment (wucke13) [reply pkgs/top-level/all-packages.nix]: Ah, TDIL. I wasn't aware of this convention. Thank you for letting me know. I updated the PR accordingly.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (lukegb) [reply pkgs/top-level/all-packages.nix:15591]: Yes, thanks! In general we add these to pkgs/top-level/aliases.nix instead - could you please make that change as well?

Comment (lukegb) [reply pkgs/top-level/all-packages.nix:15591]: Ah, was reviewing an outdated PR version...

**Pull request merged**